File: ThridYear/MA/Lab5/MobileServer/src/main/java/com/example/MobileServer/Domain/Song.java
package com.example.MobileServer.Domain;

import jakarta.persistence.*;
import lombok.*;

import java.util.Objects;

@Entity
@Table(name = "songs")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Builder
@ToString
public class Song {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private String name;

    private Integer duration;

    private String author;

    private String genre;

    private String url;

    public boolean check(Song other) {
        return Objects.equals(id, other.getId());
    }
//
//    @ManyToOne
//    @JoinColumn(name = "playlist_id", nullable = false)
//    @ToString.Exclude
//    private Playlist playlist;

}

File: ThridYear/MA/Lab2/PlaylistManager/app/src/main/java/com/example/playlistmanager/Song.kt
package com.example.playlistmanager

class Song(
    private var id: String,
    private val name: String,
    private val author: String,
    private val duration: Int,
    private val genre: String,
    private val url: String
): java.io.Serializable {

    fun getId(): String {
        return id;
    }

    fun setId(id: String) {
        this.id = id
    }

    fun getName(): String {
        return name
    }

    fun getAuthor(): String {
        return author
    }

    fun getDuration(): Int {
        return duration
    }

    fun getGenre(): String {
        return genre;
    }

    fun getUrl(): String {
        return url
    }
}